src/ctia/lib/utils.clj

```diff
@@ -134,3 +134,40 @@
 (def deep-remove-nils
   "Remove nil values from a deep nested map recursively"
   (partial deep-filter some?))
+
+(defn service-subgraph
+  "Returns a subgraph of a Trapperkeeper service graph, omitting
+  any missing entries.
+  
+  (select-services
+    {:ConfigService {:get-in-config <...>
+                     :get-config <...>}
+     :FooService {:f1 <...>
+                  :f2 <...>}
+     :BarService {:b1 <...>}}
+    :ConfigService [:get-config]
+    :FooService [:f2 :f3]
+    :MissingService [:m1])
+  ;=> {:ConfigService {:get-config <...>}
+  ;    :FooService {:f2 <...>}}
+  "
+  [graph & selectors]
+  {:pre [(map? graph)]}
+  (reduce (fn [out [service-kw fn-kws :as sel]]
+            (assert (keyword? service-kw)
+                    (pr-str service-kw))
+            (assert (every? keyword? fn-kws)
+                    (pr-str fn-kws))
+            (assert (= 2 (count sel))
+                    (str "Uneven number of selectors: "
```

Comment:
Doesn't check evenness

Reply:
Doesn't it? This checks if each element of a `(partition 2)` is of even length.

Reply:
Ah now I see why, whoops. Didn't expect `partition` to drop elements.

Reply:
Confused it with `partition-all`.